install-nhc.sh

```diff
@@ -1,28 +1,129 @@
 #!/bin/bash
 
-NHC_VERSION=1.4.3
-echo "Installed NHC verison $NHC_VERSION"
-
-wget -O nhc-$NHC_VERSION.tar.xz https://github.com/mej/nhc/releases/download/1.4.3/lbnl-nhc-1.4.3.tar.xz
-tar -xf nhc-$NHC_VERSION.tar.xz
-
-pushd lbnl-nhc-$NHC_VERSION
-
-. /etc/os-release
-case $ID in
-  ubuntu)  
-    LIBEXEDIR=/usr/lib;;
-  *) 
-    LIBEXEDIR=/usr/libexec;;
-esac
-./configure --prefix=/usr --sysconfdir=/etc --libexecdir=$LIBEXEDIR
-
-sudo make test
-echo -e "\n"
-sudo make install
-popd
+INSTALL_DIR=$1
+CUDA_DIR=$2
+
+if [[ -z "$INSTALL_DIR" ]];then
+  INSTALL_DIR=$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)
+fi
+
+if [[ -z "$CUDA_DIR" ]];then
+	CUDA_DIR=/usr/local/cuda
+fi
+
+SRC_DIR=$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)
+
+# not using src directory as insatll location
+if [ $SRC_DIR != $INSTALL_DIR ]; then
+  INSTALL_DIR=$INSTALL_DIR/azurehpc-health-checks
+fi
+
+export AZ_NHC_VERSION_LOG=$INSTALL_DIR/docs/version.log
+
+function install_lbnl_nhc(){
+  pushd $SRC_DIR/build
+  NHC_VERSION=1.4.3
+  wget -O nhc-$NHC_VERSION.tar.xz https://github.com/mej/nhc/releases/download/${NHC_VERSION}/lbnl-nhc-${NHC_VERSION}.tar.xz
+  tar -xf nhc-$NHC_VERSION.tar.xz
+  rm -f nhc-$NHC_VERSION.tar.xz
+  pushd lbnl-nhc-$NHC_VERSION
+
+  . /etc/os-release
+  case $ID in
+    ubuntu)  
+      LIBEXEDIR=/usr/lib;;
+    *) 
+      LIBEXEDIR=/usr/libexec;;
+  esac
+  ./configure --prefix=/usr --sysconfdir=/etc --libexecdir=$LIBEXEDIR
+  
+  sudo make test
+  echo -e "\n"
+  sudo make install
+  echo "NHC version: $NHC_VERSION" >> $AZ_NHC_VERSION_LOG
+  popd
+  popd
+
+}
+
+
+mkdir -p $INSTALL_DIR
+mkdir -p $INSTALL_DIR/bin
+mkdir -p $SRC_DIR/build
+mkdir -p $INSTALL_DIR/docs
+
 
-echo -e "\nRunning set up script for custom tests"
+# create version log
+AZVER=$(git describe --tags --abbrev=0)
+cat > "$AZ_NHC_VERSION_LOG" <<EOL
+This file contains the version of AzureHPC Health Checks and submodules.
+Azure-NHC: $AZVER
+submodules:
+EOL
+
+
+# install lbnl nhc
+install_lbnl_nhc 
+
+# Install NHC dependencies
+distro=`awk -F= '/^NAME/{print $2}' /etc/os-release`
+if [[ $distro =~ "Ubuntu" ]]; then
+  sudo DEBIAN_FRONTEND=noninteractive apt-get install -y libpci-dev hwloc build-essential libboost-program-options-dev libssl-dev cmake
+elif [[ $distro =~ "AlmaLinux" ]]; then
+  sudo dnf install -y pciutils-devel hwloc openssl-devel boost-devel cmake
+elif [[ $distro =~ "CentOS" ]]; then
+  sudo yum install -y  pciutils-devel hwloc openssl-devel boost-devel cmake > /dev/null
+  echo "CentOS version is not officially supported, proceed w/ caution."
+else
+  echo "OS version is not supported, Perf-test build skipped. Proceed w/ caution."
```

Comment:
Should we try and proceed w/ perf-test build just in case it happens to work but catch if it fails?

Reply:
Good idea let me address this.

Reply:
addressed